src/evaluators/SwitchStatement.js

```diff
@@ -90,32 +56,32 @@ function AbstractCaseBlockEvaluation(
       for (let i = 0; i < c.consequent.length && !(result instanceof Completion); ++i) {
         let node = c.consequent[i];
         let r = env.evaluateCompletion(node, strictCode);
+        invariant(!(r instanceof Reference));
 
-        // TODO understand Reference and how to deal with it better
         // TODO come up with a strategy for handling PossiblyNormalCompletion correctly
-        if (r instanceof Reference) {
-          throw ReportNoReferenceError();
-        } else if (r instanceof PossiblyNormalCompletion) {
-          throw ReportNoPossiblyNormalCompletionError();
+        if (r instanceof PossiblyNormalCompletion) {
+          let msg = "This operation is not yet supported on case blocks that result in a PossiblyNormalCompletion.";
```

Comment:
PossiblyNormalCompletion is an implementation detail that should not appear in a error message that is read by a developer using Prepack.

Reply:
ok. makes sense. I think your next comment has the punchline here ;-)